src/main/java/com/spring/demo/configs/IntegrationConfig.java

```diff
@@ -0,0 +1,167 @@
+  public DirectChannel itemRouterChannel() {
+    return MessageChannels.direct(Names.ITEM_ROUTER_CHANNEL).get();
+  }
+
+  @Bean(name = Names.GET_ITEMS_CHANNEL)
+  public DirectChannel getItemsChannel() {
+    return MessageChannels.direct(Names.GET_ITEMS_CHANNEL).get();
+  }
+
+  @Bean(name = Names.GET_ITEM_BY_ID_CHANNEL)
+  public DirectChannel getItemByIdChannel() {
+    return MessageChannels.direct(Names.GET_ITEM_BY_ID_CHANNEL).get();
+  }
+
+  @Bean(name = Names.ADD_NEW_ITEM_CHANNEL)
+  public DirectChannel addNewItemChannel() {
+    return MessageChannels.direct(Names.ADD_NEW_ITEM_CHANNEL).get();
+  }
+
+  @Bean(name = Names.UPDATE_ITEM_BY_ID_CHANNEL)
+  public DirectChannel updateItemByIdChannel() {
+    return MessageChannels.direct(Names.UPDATE_ITEM_BY_ID_CHANNEL).get();
+  }
+
+  @Bean(name = Names.DELETE_ITEM_BY_ID_CHANNEL)
+  public DirectChannel deleteItemByIdChannel() {
+    return MessageChannels.direct(Names.DELETE_ITEM_BY_ID_CHANNEL).get();
+  }
+
+  @Bean(name = Names.LOG_CHANNEL)
+  public DirectChannel logChannel() {
+    return MessageChannels.direct(Names.LOG_CHANNEL).get();
+  }
+
+  @Bean
+  public IntegrationFlow itemFlow() {
+    return IntegrationFlows.from(Names.INPUT_CHANNEL)
+        .handle(new ServiceActivatingHandler(loggingServiceActivator))
+        .channel(Names.ITEM_ROUTER_CHANNEL)
+        .get();
+  }
+
+  @ServiceActivator(inputChannel = Names.ITEM_ROUTER_CHANNEL)
+  @Bean
+  public HeaderValueRouter itemRouter() {
+    HeaderValueRouter itemRouter = new HeaderValueRouter(Names.API_HEADER_KEY);
+    itemRouter.setChannelMapping(Names.API_HEADER_VALUE_GET_ITEMS, Names.GET_ITEMS_CHANNEL);
+    itemRouter.setChannelMapping(Names.API_HEADER_VALUE_GET_ITEM_BY_ID,
+        Names.GET_ITEM_BY_ID_CHANNEL);
+    itemRouter.setChannelMapping(Names.API_HEADER_VALUE_ADD_NEW_ITEM, Names.ADD_NEW_ITEM_CHANNEL);
+    itemRouter.setChannelMapping(Names.API_HEADER_VALUE_UPDATE_ITEM_BY_ID,
+        Names.UPDATE_ITEM_BY_ID_CHANNEL);
+    itemRouter.setChannelMapping(Names.API_HEADER_VALUE_DELETE_ITEM_BY_ID,
+        Names.DELETE_ITEM_BY_ID_CHANNEL);
+    return itemRouter;
+  }
+
+  @ServiceActivator(inputChannel = Names.GET_ITEMS_CHANNEL)
+  public Message<List<Item>> getItemsServiceActivator(Message<String> request)
+      throws MessagingException {
+    return MessageBuilder
+        .withPayload(itemServiceDisk.getItems(request.getPayload()))
+        .copyHeaders(request.getHeaders())
+        .build();
+  }
+
+  @ServiceActivator(inputChannel = Names.GET_ITEM_BY_ID_CHANNEL)
+  public Message<Item> getItemByIdServiceActivator(Message<String> request)
+      throws MessagingException {
+
+    Item searchResult = itemServiceDisk.getItemById(request.getPayload());
+
+    if (searchResult == null) {
+      return MessageBuilder
+          .withPayload(
+              new Item(
+                  "unreal",
+                  -1,
+                  "The item you are looking for does not exist"
+              )
+          )
```

Comment:
should build payload item based on condition instead of return message by condition